tests/cupy_tests/creation_tests/test_from_data.py

```diff
@@ -475,22 +439,93 @@ def test_asfortranarray_cuda_array_zero_dim_dtype(
         return xp.asfortranarray(a, dtype=dtype_b)
 
 
-class DummyObjectWithCudaArrayInterface(object):
+max_cuda_array_interface_version = 2
+
+
+@testing.gpu
+@testing.parameterize(*testing.product({
+    'ver': tuple(range(max_cuda_array_interface_version+1)),
+    'strides': (False, None, True),
```

Comment:
I don't think `strides` has to be parameterized across all the tests.
Instead of parameterization, how about implementing separate tests, like `test_with_strides`, `test_without_strides` and `test_with_strides_none`, for example?
Appropriate version numbers should be parameterized for each of them. For example, `test_without_strides` can only be tested starting from version 2.

Reply:
We do need to test different stride configurations against different versions, and your suggestion is still a kind of parametrization. If the test coverage is good here, I prefer to keep it as is.

Reply:
The concern is `raise unittest.SkipTest` in `setUp`. If we accumulate such features as `strides` in the future, this skip condition will be more and more complicated, and more and more parameter combinations will be skipped. In other words, it's not scalable with respect to the number of features.

Reply:
@niboshi Having a long pause on this PR is perhaps a good thing --- could you read http://numba.pydata.org/numba-doc/latest/cuda/cuda_array_interface.html, and confirm that we actually don't need to skip anything in `setUp`? I think the skip reason `'ver 2+ must have the strides field'` is mistaken -- it's optional! 

The intent in https://github.com/numba/numba/pull/4609 was that we implicitly make a distinction between producers and consumers. It's suggested that producers always specify `strides`, and consumers don't have to -- they can if they want to -- check the validity of the provided info (so it's OK for `strides` to be missing, as long as it's C-contiguous).

Reply:
I roll back to a00787c. PTAL.

Reply:
I'm sorry for a long delay. 🙇
OK, let me test and merge it. Thank you for pointing out.